deepchem/transformers/__init__.py

```diff
@@ -456,3 +457,56 @@ def transform_array(self, X, y, w):
 
   def untranform(self, z):
     print("Cannot undo CoulombBinarizationTransformer.")
+
+class CDFTransformer(Transformer):
+  """Histograms the data and assigns values based on sorted list."""
+  """Acts like a Cumulative Distribution Function (CDF)."""
+  def __init__(self, transform_X=False,
+               transform_y=False,
+               bins=2,
+               dataset=None):
+    self.dataset = dataset
```

Comment:
Storing the entire dataset in transformer isn't going to work properly. The transform destructively changes dataset, so you won't have any information about the original dataset afterwards.